homeassistant/components/evohome/__init__.py

```diff
@@ -354,15 +312,39 @@ def name(self) -> str:
     @property
     def device_state_attributes(self) -> Dict[str, Any]:
         """Return the Evohome-specific state attributes."""
-        status = {}
-        for attr in self._state_attributes:
-            if attr != "setpoints":
-                status[attr] = getattr(self._evo_device, attr)
 
-        if "setpoints" in self._state_attributes:
-            status["setpoints"] = self.setpoints
+        def convert_until(status, until_key) -> str:
```

Comment:
Since you're not using `self` in any of these methods, let's move them out of being defined on each call to device_state_attributes. You can put them above the class definition.

Reply:
Sure.